components/scream/src/mct_coupling/tests/scorpio_tests.cpp

```diff
@@ -110,5 +119,122 @@ TEST_CASE("scorpio_interface_output", "") {
 
   eam_pio_finalize();
 } // TEST scorpio_interface_output
+/* ================================================================================================================ */
+TEST_CASE("scorpio_interface_input", "") {
 
+  using namespace scream;
+  using namespace scream::scorpio;
+  using ekat::util::data;
+
+  int compid=0;
+  MPI_Fint fcomm = MPI_Comm_c2f(MPI_COMM_WORLD);
+  eam_init_pio_subsystem(fcomm,compid,true);   // Gather the initial PIO subsystem data creater by component coupler
+  // Register the set of output files:
+  std::string infilename = "scorpio_output_test.nc";
+  register_infile(infilename);
+
+  const char* vec_time[] = {"time"};
+  const char* vec_x[]    = {"x"};
+  const char* vec_y[]    = {"y"};
+  const char* vec_z[]    = {"z"};
+  const char* vec_xy[]   = {"x","y"}; 
+  const char* vec_xyz[]  = {"x","y","z"};
+ 
+  register_variable(infilename,"time","time",1,vec_time, PIO_REAL,"t");
+  register_variable(infilename,"x","x-direction",1,vec_x, PIO_REAL,"x-real");
+  register_variable(infilename,"y","y-direction",1,vec_y, PIO_REAL,"y-real");
+  register_variable(infilename,"z","z-direction",1,vec_z, PIO_REAL,"z-real");
+  register_variable(infilename,"data_1d","test value for 1d field",1,vec_x, PIO_REAL,"x-real");
+  register_variable(infilename,"data_2d","test value for 2d field",2,vec_xy, PIO_REAL,"xy-real");
+  register_variable(infilename,"data_3d","test value for 3d field",3,vec_xyz, PIO_REAL,"xyz-real");
+  register_variable(infilename,"index_1d","test value for 1d field",1,vec_x, PIO_INT,"x-int");
+  register_variable(infilename,"index_2d","test value for 2d field",2,vec_xy, PIO_INT,"xy-int");
+  register_variable(infilename,"index_3d","test value for 3d field",3,vec_xyz, PIO_INT,"xyz-int");
+
+  // Create data to be written
+  std::array<Real,10> x_data;
```

Comment:
I'm confused here - this test *reads* rather than writes, right? So the comment above saying data is going to be written is wrong? And isn't the whole point of netcdf format that we can extract the variable info from the .nc file so we don't need to do the register_variable stuff above?

Reply:
I was originally expecting that this test would write a file, then read it back in. I now think that just reading an existing file is better because it pinpoints that the problem is reading rather than writing. But perhaps we should also change the test from the previous PR to use this read capability to check the output is correct in an entirely self-contained manner?

Reply:
The rest test checks the write test output. So if the output is wrong presumably it would be caught here. But we could just as easily do both steps in one test.

Reply:
The rest test checks the write test output. So if the output is wrong presumably it would be caught here. But we could just as easily do both steps in one test.